### ISSUE TITLE
Bug: eslint-plugin-graphql doesn't work fully out of the box with .graphql files

### ISSUE THREAD

**Issue by fbartho** [apollographql/eslint-plugin-graphql#109]
## Example File:

```graphql
###############################
# … redacted
# Created Date: 2018-01-25
# Author: Frederic Barthelemy
###############################
query Bootstrap {
	bootstrap @rest(path: "somePath") {
		...BootstrapId
	}
}

fragment BootstrapId on Bootstrap_ {
	id
}
```

## .eslintrc.js:
```javascript
const path = require("path");

module.exports = {
	plugins: ["graphql", "prettier", "react", "react-native"],
	parserOptions: {
		ecmaVersion: 6,
	},
	rules: {
		"prettier/prettier": "error",
		"comma-dangle": ["error", "only-multiline"],
		"graphql/template-strings": [
			"error",
			{
				env: "literal",
				schemaJsonFilepath: path.resolve(
					__dirname,
					"./dist/graphql/schema.json",
				),
			},
		],
	},
};
```

# Expected Behavior:

eslint should validate this file as passing, as it matches my schema.

# Actual Behavior:

```shell
💔  {fb/data} ~/Code/TRClient 🍣  yarn eslint . --ext .queries.graphql
yarn run v1.3.2
$ /Users/fbarthelemy/Code/TRClient/node_modules/.bin/eslint . --ext .queries.graphql
Syntax Error: Unexpected Name "ESLintPluginGraphQLFile" (1:1)
> 1 | ESLintPluginGraphQLFile`###############################
    | ^
  2 | # … redacted
  3 | # Created Date: 2018-01-25
  4 | # Author: Frederic Barthelemy
SyntaxError: Syntax Error: Unexpected Name "ESLintPluginGraphQLFile" (1:1)
> 1 | ESLintPluginGraphQLFile`###############################
    | ^
  2 | # … redacted
  3 | # Created Date: 2018-01-25
  4 | # Author: Frederic Barthelemy
    at createError (/Users/fbarthelemy/Code/TRClient/node_modules/prettier/parser-graphql.js:1:110)
    at Object.parse (/Users/fbarthelemy/Code/TRClient/node_modules/prettier/parser-graphql.js:1:876)
    at Object.parse$4 [as parse] (/Users/fbarthelemy/Code/TRClient/node_modules/prettier/index.js:24274:19)
    at formatWithCursor (/Users/fbarthelemy/Code/TRClient/node_modules/prettier/index.js:33650:27)
    at format (/Users/fbarthelemy/Code/TRClient/node_modules/prettier/index.js:33695:10)
    at Object.format (/Users/fbarthelemy/Code/TRClient/node_modules/prettier/index.js:33945:12)
    at Program (/Users/fbarthelemy/Code/TRClient/node_modules/eslint-plugin-prettier/eslint-plugin-prettier.js:378:45)
    at listeners.(anonymous function).forEach.listener (/Users/fbarthelemy/Code/TRClient/node_modules/eslint/lib/util/safe-emitter.js:47:58)
    at Array.forEach (<anonymous>)
    at Object.emit (/Users/fbarthelemy/Code/TRClient/node_modules/eslint/lib/util/safe-emitter.js:47:38)
    at NodeEventGenerator.applySelector (/Users/fbarthelemy/Code/TRClient/node_modules/eslint/lib/util/node-event-generator.js:251:26)
    at NodeEventGenerator.applySelectors (/Users/fbarthelemy/Code/TRClient/node_modules/eslint/lib/util/node-event-generator.js:280:22)
    at NodeEventGenerator.enterNode (/Users/fbarthelemy/Code/TRClient/node_modules/eslint/lib/util/node-event-generator.js:294:14)
    at CodePathAnalyzer.enterNode (/Users/fbarthelemy/Code/TRClient/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:608:23)
    at Traverser.enter [as _enter] (/Users/fbarthelemy/Code/TRClient/node_modules/eslint/lib/linter.js:1006:32)
    at Traverser._traverse (/Users/fbarthelemy/Code/TRClient/node_modules/eslint/lib/util/traverser.js:132:14)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by jnwng** [comment]
The fact that this error gets thrown in prettier and not in the GraphQL
plugin is a little suspicious. Do you mind sharing what version you’re
using for your plugins?

**Comment by fbartho** [comment]
The related dependencies are: 
```json
{
		"apollo-cache-inmemory": "1.1.x",
		"apollo-client": "2.2.x",
		"apollo-link": "1.0.x",
		"apollo-link-context": "1.0.x",
		"apollo-link-error": "1.0.x",
		"apollo-link-http": "1.3.x",
		"apollo-link-rest": "0.1.x",
		"apollo-link-state": "*",
		"eslint": "*",
		"eslint-config-prettier": "*",
		"eslint-plugin-prettier": "*",
		"eslint-plugin-react": "*",
		"eslint-plugin-react-native": "*",
		"eslint-plugin-graphql": "*",
		"prettier": "*",
		"graphql": "0.12.x",
		"graphql-tag": "2.6.x",
		"react-apollo": "^2.1.0-beta.0",
		"tslint": "*",
		"tslint-config-prettier": "*",
		"tslint-consistent-codestyle": "*",
		"tslint-eslint-rules": "*",
		"tslint-plugin-prettier": "*",
		"tslint-react": "*",
}
```
All of them should be at the latest version that my wild-cards permit. (Almost all of my deps are at latest available version).

Full [yarn.lock](https://cl.ly/2f0n00310x3W)

Do you need any specific versions @jnwng?

**Comment by eliperelman** [comment]
Related issue: https://github.com/prettier/eslint-plugin-prettier/issues/81

**Comment by jnwng** [comment]
Off the top of my head, we wrap .graphql files in a template literal
(“ESLintPluginGraphQLFile”) and proceed to do the same GraphQL linting we
do with JavaScript. However in Prettier .graphql files are just taken as-is.

I’ll poke further to see why the adjustments we make here are cascading,
but the combination of the plugins is likely the culprit.

**Comment by jnwng** [comment]
because `eslint-plugin-prettier` runs `prettier` under the hood to determine linting errors, i think its choking on `.graphql` files that should just be GraphQL and not the processed version we use here in `eslint-plugin-graphql`. im not 100% sure why processing in this plugin would affect the file being processed in a completely separate plugin (and would be helpful to generate a test case just to verify that that is in fact the case). 

let's start with that and figure out a solution afterwards — ideally we can fix the interop here and not have to poke at `eslint-plugin-prettier` or `prettier` itself

**Comment by gex** [comment]
i have the same issue with the combination of this plugin and the `eslint-plugin-prettier`.

my packages are the following:
```
"eslint": "^4.15.0",
"eslint-plugin-graphql": "^1.5.0",
"eslint-plugin-prettier": "^2.5.0",
```

**Comment by benjie** [comment]
Downgrading to eslint-plugin-prettier@2.2.0 and eslint-config-prettier@2.5.0 solves the issue for me (from 2.5.0 and 2.9.0 respectively)

**Comment by hedgepigdaniel** [comment]
According to comments here: https://github.com/prettier/eslint-plugin-prettier/issues/81

It is sufficient to downgrade eslint-plugin-prettier to 2.4.0

**Comment by mathieutu** [comment]
Hi, I'm encountering the same problem with the latest versions. 
Is something planned about that?
Thanks.

**Comment by BPScott** [comment]
I've investigated a similar issue around eslint-plugin-prettier disliking custom eslint processors and posted my findings at https://github.com/prettier/eslint-plugin-prettier/issues/81#issuecomment-424550756.

This is due to eslint-plugin-prettier trying to format preprocessed code from a graphql file (which looks like `ESLintPluginGraphQLFile\`FILE_CONTENTS_HERE\``) using its graphql processor, and then getting upset because that code is JavaScript, not graphql.

The short term fix is to disable the prettier/prettier rule for graphql files by adding an overrides section to the bottom of your eslint config, after your `rules: {}` like so:

```json
{
  "rules": {},
  "overrides": [
    {
      "files": ["*.graphql", "*.gql"],
      "rules": {
        "prettier/prettier": "off"
      }
    }
  ]
}
```

---

The longer term fix for this is to enable prettier, but forcing the babylon parser (changing ` "prettier/prettier": "off"` with ` "prettier/prettier": ["error", {parser: "babylon"}]` above). In order for that to be useful though, eslint-plugin-graphql will need to [postprocess its results properly](https://github.com/apollographql/eslint-plugin-graphql/blob/master/src/index.js#L516-L518) instead of supressing all issue that aren't related to its own rules.

**Comment by fbartho** [comment]
@BPScott -- looking at your linked issues, is this issue resolved now?

**Comment by BPScott** [comment]
@fbartho yes, as of the latest version of eslint-plugin-prettier it contains a [list of file extensions to force the parser for](https://github.com/prettier/eslint-plugin-prettier/blob/master/eslint-plugin-prettier.js#L177-L202) which means that my talk of "forcing the babylon parser" is already handled for you within eslint-plugin-prettier.

Updating to the latest version eslint-plugin-prettier should fix the original issue.

**Comment by fbartho** [comment]
I can report that I did get past this error today!

Now unfortunately, I need to change stuff about our process in order to document KnownDirectives and other things, which is less than obvious. But thanks for the confirmation @BPScott!
